### PR TITLE
Add support for gitea. WIP

### PR DESCRIPTION
**Pull request by C-J1**
Not quite ready to be signed and sealed. Raised the PR to share the WIP

Issue #3

### CODE REVIEW

File: group_vars/all.yml.dist [hunk 1]
```diff
@@ -31,6 +31,7 @@ stats_enabled: false
 duplicati_enabled: true
 nextcloud_enabled: false
 nginx_enabled: true # if you enable nginx, you'll need to map ports 80 and 443 on your router to your Ansible-NAS
+gitea_enable: false
```
Comment (davestephens) [inline group_vars/all.yml.dist]: Should be ```gitea_enabled```

File: tasks/gitea.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+- name: Create Gitea container
+  docker_container:
+    name: gitea
+    image: gitea/gitea:1.6
+    pull: true
+    links:
+        - gitea-mysql:db
+    volumes:
+      - gitea_data:/data
```
Comment (davestephens) [inline tasks/gitea.yml]: Interested on why you've chosen volumes over bind mounts?

**Comment by davestephens** [comment]
This is awesome, thank you! I'll pull and test tomorrow :+1:

File: tasks/gitea.yml (hunk 2 above)
Comment (C-J1) [reply tasks/gitea.yml]: No particular reason. I've done a bunch of reading on volumes vs bind mounts, and everything seems to be "use volumes, they are awesome", except if you want to access the content outside of the container (via, for example, a samba share :D).

With bind mounts, you get the permissions issues, but more flexibility. I'm happy to follow your standard of bind mounts, but I'd like to look at the permissions side of things. Similar to the way that FreeNAS jails handled it (create a app specific user on the host, give it permissions based on what it needs to access (create groups like movieusers or mediausers for example), and then pass in the UID/GID (where appropriate) to the container.

Thoughts?

**Comment by C-J1** [comment]
I swapped to bind mounts for this. It does make sense to be consistent across all the containers.

I hope you don't mind the git user and group accounts. I felt that it makes it a bit easier to allow access to app conf and data directories by letting normal users be added to the group. Something I'd like to replicate across the other containers (eventually). It also stops the apps being run as the root user inside their containers.

**Comment by davestephens** [comment]
The main reason I prefer bind mounts is that you know where your data will actually be on your filesystem, making backups, troubleshooting, general nosiness etc easier!

Sorting the permissions properly has been on my list for ages, just not got to it yet, so this is perfect :+1:

**Comment by davestephens** [comment]
@C-J1 This is awesome dude, thank you! (actually gonna use this one myself too :+1:)

**Pull request merged**